### PR TITLE
move GPS aggregation and blending to sensors

### PR DESCRIPTION
**Pull request by dagar**
This moves the multi-GPS handling (and optional blending) from the EKF2 frontend (ekf2_main) to the sensors hub module.

The new module within sensors consumes all `sensor_gps` messages (formerly vehicle_gps_position) from the GPS driver or UAVCAN, etc. Then based on configuration either passes on the first or blends multiple to create a single `vehicle_gps_position` that the rest of the system consumes.

Although technically in the sensors module, this is a separate work queue item that's scheduled by any sensor_gps publication.

### Changes
 - GPS producers now publish `sensor_gps`
 - all GPS consumers now get the selected primary or blended GPS state `vehicle_gps_position`
 - Parameter `EKF2_GPS_MASK` -> `SENS_GPS_MASK`
 - Parameter `EKF2_GPS_TAU` -> `SENS_GPS_TAU`

### CODE REVIEW

File: msg/sensor_gps.msg [hunk 1]
```diff
@@ -1,19 +1,36 @@
-# EKF blended position in WGS84 coordinates.
+# GPS position in WGS84 coordinates.
```
Comment (mhkabir) [inline msg/sensor_gps.msg]: Let's fix the things brought up in https://github.com/PX4/Firmware/issues/6275 now.

**Comment by dagar** [comment]
This should also make it easier to add unit tests for GPS blending.

File: msg/sensor_gps.msg (hunk 1 above)
Comment (dagar) [reply msg/sensor_gps.msg]: Yes this would probably be the time.

File: src/modules/ekf2/ekf2_main.cpp [hunk 2]
```diff
@@ -924,102 +857,33 @@ void Ekf2::Run()
...
-				gps.heading_offset = _gps_output[_gps_select_index].yaw_offset;
-				gps.selected = _gps_select_index;
-
-				// Publish to the EKF blended GPS topic
-				_blended_gps_pub.publish(gps);
-
-				// clear flag to avoid re-use of the same data
-				_gps_new_output_data = false;
+			if (_vehicle_gps_position_sub.copy(&gps)) {
+				gps_message gps_msg{};
```
Comment (bresch) [inline src/modules/ekf2/ekf2_main.cpp]: I would still like to use a function such as `fillGpsMsgWithVehicleGpsPosData` (that you forgot to remove) to reduce the amount of code in this massive function.

Comment (dagar) [reply src/modules/ekf2/ekf2_main.cpp]: Thanks, I need to take another look. I believe this PR predates that ekf2 refactor.

**Comment by dakejahl** [comment]
Continued in
https://github.com/PX4/Firmware/pull/14447

**Pull request closed without merging**